mapproxy/cache/tile.py

```diff
@@ -465,9 +465,17 @@ def get_map_from_source(source):
                 return (img, source.coverage)
 
         layers = []
+        transparent_layers = []
         for layer in async_.imap(get_map_from_source, self.sources):
             if layer[0] is not None:
                 layers.append(layer)
+                try:
+                    transparent_layers.append(layer[0].image_opts.transparent)
+                except AttributeError:
+                    transparent_layers.append(None)
+
+        if len(transparent_layers) > 0 and all(transparent_layers):
+            self.tile_mgr.image_opts.transparent = True
```

Comment:
```suggestion
        all_transparent = True
        for layer in async_.imap(get_map_from_source, self.sources):
            if layer[0] is not None:
                layers.append(layer)
                if 'image_opts' in layer[0] and layer[0].image_opts.transparent:
                     all_transparent = all_transparent and layer[0].image_opts.transparent
                else: 
                     all_transparent = False

        if all_transparent:
            self.tile_mgr.image_opts.transparent = True
```

please check if this works. The code is a little clearer than using `except`.

Reply:
Thanks for your feedbacks, and thanks again @tobwen for this code snippet. It works perfectly fine as is but we plan to test your suggestion Simon. Will let you know in a few days.

Reply:
I think this one was it: https://github.com/tobwen/mapproxy/commit/cbc3ccaee92996fccb23a51873038e4c462aca23

I hope I can test it tomorrow.